torch_frame/config/image_embedder.py

```diff
@@ -0,0 +1,29 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Callable
+
+from torch import Tensor
+
+
+@dataclass
+class ImageEmbedderConfig:
+    r"""Image embedder model that maps a list of images into PyTorch
+    Tensor embeddings.
+
+    Args:
+        image_embedder (callable): A callable image embedder that takes a
+            list of path to images as input and outputs the PyTorch Tensor
+            embeddings for that list of images.
+        batch_size (int, optional): Batch size to use when encoding the
+            images. If set to :obj:`None`, the image embeddings will
+            be obtained in a full-batch manner. (default: :obj:`None`)
+
+    """
+    # TODO (zecheng): Add more types here, such as list of Image or pixels
+    image_embedder: Callable[[list[str]], Tensor]
```

Comment:
For this, I feel a better approach is to have an additional attribute
```python
image_retriever: Callable[[list[sr]], list[Image]]
image_embedder: Callable[list[Image], Tensor]
```
Basically, let `image_retriever` generate `Image` object, which is then consumed by the image embedding model.